package/MDAnalysis/analysis/parallel.py

```diff
@@ -0,0 +1,375 @@
+    >>> from MDAnalysis.analysis.rms import RMSD
+    >>> R = RMSD(...) # setup the run
+    >>> n_workers = 2
+    >>> backend = ThreadsBackend(n_workers=n_workers)
+    >>> R.run(backend=backend)
+
+    .. versionadded:: 2.7.0
+    """
+
+    def __init__(self, n_workers: int):
+        self.n_workers = n_workers
+        self._validate()
+
+    def _get_checks(self):
+        """Get dictionary with `condition: error_message` pairs that ensure the validity of the backend instance
+
+        Returns
+        -------
+        dict
+            dictionary with `condition: error_message` pairs that will get checked during _validate() run
+
+        .. versionadded: 2.7.0
+        """
+        return {
+            isinstance(self.n_workers, int)
+            and self.n_workers > 0: f"n_workers should be positive integer, got {self.n_workers=}",
+        }
+
+    def _get_warnings(self):
+        """Get dictionary with `condition: warning_message` pairs that ensure the good usage of the backend instance
+
+        Returns
+        -------
+        dict
+            dictionary with `condition: warning_message` pairs that will get checked during _validate() run
+
+        .. versionadded: 2.7.0
+        """
+        return dict()
+
+    def _validate(self):
+        """Check correctness (e.g. `dask` is installed if using `backend='dask'`)
+        and good usage (e.g. `n_workers=1` if backend is serial) of the backend
+
+        Raises
+        ------
+        ValueError
+            if one of the conditions in :meth:`self._get_checks()` is True
+
+        .. versionadded: 2.7.0
+        """
+        for check, msg in self._get_checks().items():
+            if not check:
+                raise ValueError(msg)
+        for check, msg in self._get_warnings().items():
+            if not check:
+                warnings.warn(msg)
+
+    def apply(self, func: Callable, computations: list) -> list:
+        """Main function that will get called when using an instance of an object, mapping function to all tasks
+        in the `computations` list. Should effectively be equivalent to running [func(item) for item in computations]
+        while using the parallel backend capabilities.
+
+        Parameters
+        ----------
+        func : Callable
+            function to be called on each of the tasks in computations list
+        computations : list
+            computation tasks to apply function to
+
+        Returns
+        -------
+        list
+            list of results of the function
+
+        .. versionadded: 2.7.0
+        """
+        raise NotImplementedError("Should be re-implemented in subclasses")
+
+
+class BackendSerial(BackendBase):
+    """A built-in backend that does serial execution of the function, without any parallelization
+
+    .. versionadded: 2.7.0
+    """
+
+    def _get_warnigns(self):
+        return {self.n_workers > 1, "n_workers > 1 will be ignored while executing with backend='serial'"}
```

Comment:
```suggestion
    def _get_warnigns(self):
        return {self.n_workers > 1, "n_workers is ignored when executing with backend='serial'"}
```
The `>1` part might be a bit confusing for the end user. I'd just say that `n_workers` is ignored (unless it is 1, for which no warning is issued).